Template/Tag/MatomoTag.web.js

```diff
@@ -21,8 +21,15 @@
     // We use the stringify and parse to make sure that we have a copy and not a reference
     var localPaq = window._paq && window._paq.length ? JSON.parse(JSON.stringify(window._paq)) : localPaq || [];
     var indexesOfConfigs = [];
+    var hasTrackPageViewViaPaq = false;
+    var j = -1;
     // Clear window._paq to prevent things from being tracked too early
     while (window._paq.length > 0) {
+        j++;
+        // Check if trackPageView is set in _paq variable to ensure tracking works even with no callback
+        if (Array.isArray(window._paq[j]) && window._paq[j].toString() == ['trackPageView'].toString()) {
+            hasTrackPageViewViaPaqIndex = true;
+        }
         window._paq.pop();
```

Comment:
I think what you have would be checking elements at the beginning while the elements are being removed from the end, which means that the end elements wouldn't be checked. How about something like the following?
```suggestion
    var hasTrackPageViewViaPaq = false;
    // Clear window._paq to prevent things from being tracked too early
    while (window._paq.length > 0) {
        var poppedItem = window._paq.pop();
        // Check if trackPageView is set in _paq variable to ensure tracking works even with no callback
        if (Array.isArray(poppedItem) && poppedItem.toString() == ['trackPageView'].toString()) {
            hasTrackPageViewViaPaqIndex = true;
        }
```

Reply:
@snake14 makes sense :+1:

Reply:
If this change makes sense and you apply it, I approve of this PR :+1: